src/State/ProviderInterface.php

```diff
@@ -20,7 +20,7 @@
 /**
  * Retrieves data from a persistence layer.
  *
- * @template T of object
+ * @template T of object|array
```

Comment:
shouldn't it be iterable instead of array ?

Reply:
right!

Reply:
in fact, no, it's not an iterable, as it's really an array of compound datas.
Like, in my case, in Doctrine Orm, I have a DQL query like `SELECT o, some_scalar as somevalue ...` which produce an array with the object/entity in the first array element, and the scalar in 'somevalue' array element.

So it's not an iterable.